File: php/src/Advertisements/Advertisement/Domain/ValueObjects/Description.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\Advertisement\Domain\ValueObjects;

use Demo\App\Advertisements\Advertisement\Domain\Exceptions\DescriptionEmptyException;
use Demo\App\Advertisements\Advertisement\Domain\Exceptions\DescriptionTooLongException;

final class Description
{
    public function __construct(private string $value)
    {
        $this->validate($value);
    }

    public function value(): string
    {
        return $this->value;
    }

    /**
     * @throws DescriptionTooLongException
     * @throws DescriptionEmptyException
     */
    private function validate(string $value): void
    {
        if (mb_strlen($value) === 0) {
            throw DescriptionEmptyException::build();
        }

        if (mb_strlen($value) > 200) {
            throw DescriptionTooLongException::withLongitudeMessage($this->value);
        }
    }
}

File: kotlin/src/main/kotlin/advertisements/advertisement/domain/value_object/Description.kt
package advertisements.advertisement.domain.value_object

import advertisements.advertisement.domain.exceptions.DescriptionEmptyException
import advertisements.advertisement.domain.exceptions.DescriptionTooLongException

class Description(private var value: String) {

    init {
        this.validate(value)
    }

    fun value(): String {
        return value
    }

    private fun validate(value: String) {
        if (value.isEmpty()) {
            throw advertisements.advertisement.domain.exceptions.DescriptionEmptyException.build()
        }

        if (value.length > 200) {
            throw DescriptionTooLongException.withLongitudeMessage(value)
        }
    }
}
